jupyterlab3/entrypoint.sh

```diff
@@ -156,6 +156,7 @@ MEMORY=$(get_max_memory "$NAMESPACE")
 # TBD maap-py install
 
 source /opt/conda/bin/activate base
+conda config --set solver libmamba
```

Comment:
I want to learn why this line needed to be added to the entrypoint, in addition to the line that already exists in our base dockerfile?

Reply:
My guess was only this file could create a file within the workspace? Do you think that could be true?

Reply:
@grallewellyn 

> My guess was only this file could create a file within the workspace? Do you think that could be true?

We can create files via commands entered in our dockerfiles. Is there an issue you ran into where something didn't work as expected? Moving the `conda config` to the entrypoint shouldn't behave any different than what we currently have, but conda has surprised as many other times.

> We also could probably remove all the `conda config --set solver libmamba` from all the base image dockerfiles

If we do so, it means we should definitely be explicit about what solver we use when running `conda install` or `conda env update`.

Reply:
> Is there an issue you ran into where something didn't work as expected?

I guess the issue would be that `conda config --set solver libmamba` in the dockerfile isn't creating the `.condarc` file. I didn't catch onto this before because once the `.condarc` file is created (which happened for me by locally running `conda config --set solver libmamba`), libmamba is always set even when creating new workspaces. The `.condarc` file needs to be removed each time for proper testing

> If we do so, it means we should definitely be explicit about what solver we use when running `conda install` or `conda env update`.

I agree

Reply:
> > Is there an issue you ran into where something didn't work as expected?
> 
> I guess the issue would be that `conda config --set solver libmamba` in the dockerfile isn't creating the `.condarc` file. I didn't catch onto this before because once the `.condarc` file is created (which happened for me by locally running `conda config --set solver libmamba`), libmamba is always set even when creating new workspaces. The `.condarc` file needs to be removed each time for proper testing

Check out the `--system` parameter, may help you: https://docs.conda.io/projects/conda/en/latest/commands/config.html#Config%20File%20Location%20Selection

Also worthy to reference: https://conda.io/projects/conda/en/latest/user-guide/configuration/use-condarc.html#searching-for-condarc

@grallewellyn

Reply:
(fyi) updated those links for conda 22.11.x

Reply:
> Also worthy to reference: https://conda.io/projects/conda/en/latest/user-guide/configuration/use-condarc.html#searching-for-condarc

This is interesting because it makes it seem like the .condarc file could just be stored in a different location from running `conda config --set solver libmamba` in the base image dockerfile, but I am not seeing it at all 

I tried changing the vanilla/dockerfile to `conda config --system --set solver libmamba` and .condarc is now being created on launch in /opt/conda/ and from running with `--verbose` it seems like libmamba solver is being used 

@anilnatha Do we want to add the `--system` tag to all base image dockerfiles instead of running in entrypoint.sh?

Reply:
> Do we want to add the --system tag to all base image dockerfiles instead of running in entrypoint.sh? 

Yes, especially since it sounds like adding that parameter worked. We can always move the conda config to entrypoint.sh later if there is a compelling reason to do so.

Reply:
Okay, building for all base images, will do some testing then update
https://repo.dit.maap-project.org/root/maap-workspaces/-/pipelines/4314

Reply:
>This is interesting because it makes it seem like the .condarc file could just be stored in a different location from running conda config --set solver libmamba in the base image dockerfile, but I am not seeing it at all

I ran into similar issues when trying to get environment variables to work at the user level vs at the system level. There are nuances with users and their home directory that I don't fully understand yet with respect to the docker build process.

Reply:
This works with all images except isce2 and rgedi. rgedi workspace builds, but all conda commands give this error
```
Error while loading conda entry point: conda-libmamba-solver (libarchive.so.19: cannot open shared object file: No such file or directory)
```
But since we deprecated rgedi I think this is fine
I verified that the .condarc file was created and libmamba was being used in installs for vanilla, pangeo, r and isce3

@anilnatha I think you have to approve again if you agree